webapp/client/src/pages/TaxNumberPage.test.js

```diff
@@ -0,0 +1,402 @@
+      ).toBeNull();
+    });
+
+    describe("Select Baden-Württemberg", () => {
+      beforeEach(() => {
+        userEvent.selectOptions(
+          screen.getByLabelText("Bundesland", notExactQueryOptions),
+          ["bw"]
+        );
+      });
+
+      it("Should show question for tax number", () => {
+        expect(
+          screen.queryByText(
+            "Haben Sie bereits eine Steuernummer",
+            notExactQueryOptions
+          )
+        ).not.toBeNull();
+        expect(screen.queryByText("Ja")).not.toBeNull();
+        expect(screen.queryByText("Nein")).not.toBeNull();
+      });
+
+      it("Should show state selection with correct selected value", () => {
+        expect(
+          screen.queryByLabelText("Bundesland", notExactQueryOptions)
+        ).not.toBeNull();
+        expect(screen.queryAllByRole("combobox")[0]).toHaveValue("bw");
+      });
+
+      it("Do not show tax number input", () => {
+        expect(screen.queryByText("Steuernummer")).toBeNull();
+      });
+
+      it("Should show tax office selection", () => {
+        expect(
+          screen.queryByLabelText(
+            "Wählen Sie Ihr Finanzamt",
+            notExactQueryOptions
+          )
+        ).not.toBeNull();
+      });
+
+      it("Should not show confirmation field", () => {
+        expect(
+          screen.queryByLabelText("Hiermit bestätige", notExactQueryOptions)
+        ).toBeNull();
+      });
+
+      describe("Select tax office", () => {
+        beforeEach(() => {
+          userEvent.selectOptions(
+            screen.getByLabelText("Finanzamt", notExactQueryOptions),
+            ["2801"]
+          );
+        });
+
+        it("Should show question for tax number", () => {
+          expect(
+            screen.queryByText(
+              "Haben Sie bereits eine Steuernummer",
+              notExactQueryOptions
+            )
+          ).not.toBeNull();
+          expect(screen.queryByText("Ja")).not.toBeNull();
+          expect(screen.queryByText("Nein")).not.toBeNull();
+        });
+
+        it("Should show state selection with correct selected value", () => {
+          expect(
+            screen.queryByLabelText("Bundesland", notExactQueryOptions)
+          ).not.toBeNull();
+          expect(screen.queryAllByRole("combobox")[0]).toHaveValue("bw");
+        });
+
+        it("Should not show tax number input", () => {
+          expect(screen.queryByText("Steuernummer")).toBeNull();
+        });
+
+        it("Should show tax office selection", () => {
+          expect(
+            screen.queryByLabelText(
+              "Wählen Sie Ihr Finanzamt",
+              notExactQueryOptions
+            )
+          ).not.toBeNull();
+        });
+
+        it("Should show confirmation field", () => {
+          expect(
+            screen.queryByLabelText("Hiermit bestätige", notExactQueryOptions)
+          ).not.toBeNull();
+        });
```

Comment:
Should we in the end click on "next page" and that there are no errors? Or is that even possible with this test? 🤔

Reply:
It is not. Because we only test the component there is nothing to post to and the errors are coming from the backend